src/classes/TDTM_Runnable_TEST.cls

```diff
@@ -36,65 +36,132 @@
+        dmlWrapper.objectsToInsert.addAll(objectsToInsert);
+        dmlWrapper.objectsToUpdate.addAll(objectsToUpdate);
+
+        dmlWrapper.groupByType();
+
+        System.assertEquals(3, dmlWrapper.objectsToInsert.size(),
+            'Duplicate records should be removed from insert list: ' + dmlWrapper.objectsToInsert);
+        System.assertEquals(3, dmlWrapper.objectsToUpdate.size(),
+            'Duplicate records should be removed from update list: ' + dmlWrapper.objectsToUpdate);        
+
+        update dmlWrapper.objectsToUpdate;
+    }
     
-    	List<Contact> contacts = new List<Contact>();
-		for(integer i = 0; i< 200; i++) {
-		    Contact contact = new contact(firstname='Test', lastname='Person' + i);
-		    contact.MailingStreet = 'multiple-address';
-		    contact.MailingCity = 'Cupertino';
-		    contact.MailingState = 'California';
-		    contact.MailingPostalCode = '95014';
-		    contact.MailingCountry = 'United States';
-		    contacts.add(contact);
-		}
-		
-		Test.startTest();
-		insert contacts;
-		Test.stopTest();
-		
-		Integer errors = [select count() from Error__c];
-		list<Error__c> listErr = [select Full_Message__c, Object_Type__c, Record_URL__c from Error__c];
-		system.debug('**** listErr: ' + listErr);
-		System.assertEquals(0, errors);
+    /*************************************************************************** 
+    * @description Verifies records are grouped by their sObjectType in order to
+    * prevent "Cannot have more than 10 chunks in a single operation" error.
+    **/
+    @isTest
+    private static void groupByTypeShouldGroupSObjectsByType() {
+        List<Id> accountIds = extractIds([SELECT Id FROM Account]);
+
+        // Assign more than 10 groups of sObjectTypes in sequence
+        List<SObject> objectsToInsert = new List<SObject>{
+            new Account(Name = 'GroupByAccountA'),
+            new Address__c(Household_Account__c = accountIds[0]),
+            new Contact(LastName = 'GroupBySmith', AccountId = accountIds[0]),
+            new Account(Name = 'GroupByAccountB'),
+            new Address__c(Household_Account__c = accountIds[1]),
+            new Contact(LastName = 'GroupByClark', AccountId = accountIds[1]),
+            new Account(Name = 'GroupByAccountC'),
+            new Address__c(Household_Account__c = accountIds[2]),
+            new Contact(LastName = 'GroupByJacobsen', AccountId = accountIds[2]),
+            new Account(Name = 'GroupByAccountD'),
+            new Contact(LastName = 'GroupByJackson', AccountId = accountIds[2])
+        };
+
+        TDTM_Runnable.DmlWrapper dmlWrapper = new TDTM_Runnable.DmlWrapper();
+        dmlWrapper.objectsToInsert.addAll(objectsToInsert);
+
+        dmlWrapper.groupByType();
+
+        System.assertEquals(objectsToInsert.size(), dmlWrapper.objectsToInsert.size(),
+            'Insert list size should be unchanged: ' + dmlWrapper.objectsToInsert);
+
+        insert dmlWrapper.objectsToInsert;
```

Comment:
Yes, the "duplicate id in the list" testing in this test, and "cannot have more than 10 chunks in a single operation" in the next test is verified by actual insert, ie update. If the `dmlWrapper.groupByType()` is commented out, the insert/update will throw an exception and the unit test would fail.

The [previous unit test version](https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/TDTM_Runnable_TEST.cls#L74) actually did not throw an exception when [grouping sObjects by type](https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/TDTM_Runnable.cls#L157) is commented out, meaning we do not have to insert 210 Contacts to verify `TDTM_Runnable` methods.

Reply:
Makes sense. Nice solution to this.